packages/react-component-library/src/components/NumberInput/NumberInput.tsx

```diff
@@ -87,6 +87,12 @@ export const NumberInput: React.FC<NumberInputProps> = ({
     setNextValue,
   } = useValue(value)
 
+  // if something changes the value, such as a call to setField in Formik
```

Comment:
It feels like these comments are not required as this is a pattern we have across the codebase.